pkg/v1/config/clientconfig.go

```diff
@@ -742,9 +744,31 @@ func GetDiscoverySources(serverName string) []configv1alpha1.PluginDiscovery {
 		}
 		discoverySources = append(discoverySources, defaultClusterK8sDiscovery)
 	}
+
+	// If the current server type is global, then add the default REST endpoint
+	// for the discovery service
+	if server.Type == configv1alpha1.GlobalServerType {
+		defaultRestDiscovery := configv1alpha1.PluginDiscovery{
+			REST: &configv1alpha1.GenericRESTDiscovery{
+				Name:     fmt.Sprintf("default-%s", serverName),
+				Endpoint: appendURLScheme(server.GlobalOpts.Endpoint),
+				BasePath: "v1alpha1/system/binaries/plugins",
+			},
+		}
+		discoverySources = append(discoverySources, defaultRestDiscovery)
+	}
```

Comment:
As part of https://github.com/vmware-tanzu/tanzu-framework/issues/1764 we are planning to decouple default plugin discovery from core tanzu cli.

So, I think we should not add this code here for discovering default TMC plugins. Instead we should use some other mechanism. Like `tanzu plugin source add` command to add discovery source after installing core cli.
cc @vuil

Reply:
Will give that a shot, so basically i would want the tmc-plugins to be discovered and installed on running `tanzu plugin sync`. If thats still achievable if i go the suggested route? @anujc25

Reply:
Yes. That will still be possible. Basically `tanzu plugin source add` will add the same discovery source that we are adding here. But we don't want to hardcode any default behavior of adding discovery source as part of core tanzu cli.
Because, different customers would like to include only specific product plugins. And that is the reason behind decoupling default standalone plugins from core cli as part of https://github.com/vmware-tanzu/tanzu-framework/issues/1764.

Reply:
Gotcha. Thanks for clarifying 👍

Reply:
@pgandigesang sorry I think I misunderstood this with the standalone plugins. This looks like we are identifying plugins based on the context and this might still be applicable.